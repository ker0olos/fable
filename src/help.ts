import i18n from '~/src/i18n.ts';
import user from '~/src/user.ts';

import * as discord from '~/src/discord.ts';

import {
  COSTS,
  MAX_PULLS,
  RECHARGE_DAILY_TOKENS_HOURS,
  RECHARGE_MINS,
  STEAL_COOLDOWN_HOURS,
} from '~/db/index.ts';

import { PARTY_PROTECTION_PERIOD } from '~/src/steal.ts';

function pages({
  userId,
  index,
}: {
  userId: string;
  index: number;
}): discord.Message {
  const locale = user.cachedUsers[userId]?.locale;

  const pages = [
    new discord.Message()
      .addComponents([
        new discord.Component().setId('gacha', userId).setLabel('/gacha'),
        new discord.Component().setId('now', userId).setLabel('/now'),
        new discord.Component()
          .setLabel('GitHub')
          .setUrl('https://github.com/ker0olos/fable'),
        new discord.Component()
          .setLabel(i18n.get('support-server', locale))
          .setUrl('https://discord.gg/H69RVBxeYY'),
        new discord.Component()
          .setLabel(i18n.get('donate', locale))
          .setUrl('https://www.buymeacoffee.com/ker0olos'),
      ])
      .addEmbed(
        new discord.Embed()
          .setAuthor({ name: '1.' })
          .setTitle('`/gacha`')
          .setDescription(
            [
              i18n.get('help-page1-before-anything', locale),
              '',
              i18n.get(
                'help-page1-you-get-n-every-t',
                locale,
                RECHARGE_MINS,
                MAX_PULLS
              ),
              '',
              i18n.get('help-page1-characters-are-exclusive', locale),
              '',
              i18n.get(
                'help-page1-characters-vary',
                locale,
                discord.emotes.smolStar
              ),
              '',
              i18n.get('help-page1-footer', locale),
              discord.empty,
            ].join('\n')
          )
          .setFooter({ text: `${i18n.get('aliases', locale)}: /w, /q` })
      ),
    new discord.Message().addEmbed(
      new discord.Embed()
        .setAuthor({ name: '2.' })
        .setTitle('`/merge`')
        .setDescription(
          [
            i18n.get('help-page2-sacrifice-characters', locale),
            '',
            i18n.get('help-page2-merge-protect', locale),
            '',
            i18n.get('help-page2-merging', locale),
            '',
            i18n.get('help-page2-auto-merging', locale),
            '',
            i18n.get(
              'help-page2-merge-example-1',
              locale,
              discord.emotes.smolStar
            ),
            i18n.get(
              'help-page2-merge-example-2',
              locale,
              discord.emotes.smolStar
            ),
            '',
            i18n.get('help-page2-footer', locale),
            discord.empty,
          ].join('\n')
        )
        .setFooter({ text: `${i18n.get('aliases', locale)}: /synthesis` })
    ),
    new discord.Message()
      .addComponents([
        new discord.Component()
          .setId('cstars', '5', userId)
          .setLabel('/collection stars 5'),
        new discord.Component()
          .setId('cstars', '4', userId)
          .setLabel('/collection stars 4'),
      ])
      .addEmbed(
        new discord.Embed()
          .setAuthor({ name: '3.' })
          .setTitle('`/party`')
          .setDescription(
            [
              i18n.get('help-page3-party-required', locale),
              '',
              i18n.get('help-page3-party-presets-disclaimer', locale),
              '',
              i18n.get('help-page3-party-view', locale),
              '',
              i18n.get('help-page3-party-assign-1', locale),
              i18n.get('help-page3-party-assign-2', locale),
              '',
              i18n.get('help-page3-footer', locale),
              discord.empty,
            ].join('\n')
          )
          .setFooter({ text: `${i18n.get('aliases', locale)}: /team, /p` })
      ),
    new discord.Message().addEmbed(
      new discord.Embed()
        .setAuthor({ name: i18n.get('help-page4-title', locale) })
        .setDescription(
          [
            i18n.get('help-page4-negotiations', locale),
            '',
            i18n.get('help-page4-party-is-protected', locale),
            '',
            i18n.get('help-page4-inactive-users', locale),
            '',
            i18n.get(
              'help-page4-losing-protection',
              locale,
              PARTY_PROTECTION_PERIOD
            ),
            '',
            i18n.get(
              'help-page4-steal-chance',
              locale,
              discord.emotes.smolStar
            ),
            '',
            i18n.get('help-page4-cooldown', locale, STEAL_COOLDOWN_HOURS / 24),
            '',
            i18n.get('help-page4-footer', locale),
            discord.empty,
          ].join('\n')
        )
    ),
    new discord.Message()
      .addComponents([
        new discord.Component()
          .setId('buy', 'bguaranteed', userId, '5')
          .setLabel(`/buy guaranteed 5`),
      ])
      .addEmbed(
        new discord.Embed()
          .setAuthor({ name: i18n.get('help-page5-title', locale) })
          .setDescription(
            [
              i18n.get('help-page5-shop', locale),
              '',
              i18n.get(
                'help-page5-daily-tokens',
                locale,
                RECHARGE_DAILY_TOKENS_HOURS
              ),
              '',
              i18n.get('help-page5-buy-normal', locale, RECHARGE_MINS),
              '',
              i18n.get('help-page5-buy-guaranteed', locale),
              '',
              i18n.get('help-page5-example', locale, discord.emotes.smolStar),
              '',
              i18n.get(
                'help-page5-buy-3s',
                locale,
                discord.emotes.smolStar,
                COSTS.THREE
              ),
              i18n.get(
                'help-page5-buy-4s',
                locale,
                discord.emotes.smolStar,
                COSTS.FOUR
              ),
              i18n.get(
                'help-page5-buy-5s',
                locale,
                discord.emotes.smolStar,
                COSTS.FIVE
              ),
              '',
              i18n.get('help-page5-footer', locale),
              discord.empty,
            ].join('\n')
          )
          .setFooter({ text: `${i18n.get('aliases', locale)}: /shop` })
      ),
    new discord.Message().addEmbed(
      new discord.Embed()
        .setAuthor({ name: i18n.get('help-page7-title', locale) })
        .setDescription(
          [
            `- \`/gacha\` \`/w\`: _${i18n.get('/gacha', locale)}_`,
            `- \`/q\`: _${i18n.get('/q', locale)}_`,
            `- \`/pull\` \`/guaranteed\`: _${i18n.get('/pull', locale)}_`,
            '',
            `- \`/now\` \`/tu\`: _${i18n.get('/now', locale)}_`,
            `- \`/search\` \`/anime\` \`/manga\` \`/series\`: _${i18n.get(
              '/search',
              locale
            )}_`,
            `- \`/character\` \`/char\`: _${i18n.get('/character', locale)}_`,
            '',
            `- \`/merge\` \`/synthesize\`: _${i18n.get('/merge', locale)}_`,
            '',
            `- \`/party view\` \`/team view\` \`/p view\`: _${i18n.get(
              '/party view',
              locale
            )}_`,
            `- \`/party assign\` \`/team assign\` \`/p assign\`: _${i18n.get(
              '/party assign',
              locale
            )}_`,
            `- \`/party swap\` \`/team swap\` \`/p swap\`: _${i18n.get(
              '/party swap',
              locale
            )}_`,
            `- \`/party remove\` \`/team remove\` \`/p remove\`: _${i18n.get(
              '/party remove',
              locale
            )}_`,
            `- \`/party clear\` \`/team clear\` \`/p clear\`: _${i18n.get(
              '/party clear',
              locale
            )}_`,
            '',
            `- \`/collection show\` \`/coll show\` \`/mm show\`: _${i18n.get(
              '/coll show',
              locale
            )}_`,
            `- \`/collection stars\` \`/coll stars\` \`/mm stars\`: _${i18n.get(
              '/coll stars',
              locale
            )}_`,
            `- \`/collection media\` \`/coll media\` \`/mm media\`: _${i18n.get(
              '/coll media',
              locale
            )}_`,
            `- \`/collection sum\` \`/coll sum\` \`/mm sum\`: _${i18n.get(
              '/coll sum',
              locale
            )}_`,
            '',
            `- \`/steal\`: _${i18n.get('/steal', locale)}_`,
            `- \`/trade\` \`/offer\`: _${i18n.get('/trade', locale)}_`,
            `- \`/give\` \`/gift\`: _${i18n.get('/give', locale)}_`,
          ].join('\n')
        )
    ),
    new discord.Message().addEmbed(
      new discord.Embed()
        .setAuthor({ name: i18n.get('help-page8-title', locale) })
        .setDescription(
          [
            `- \`/nick\`: _${i18n.get('/nick', locale)}_`,
            `- \`/image\` \`/custom\`: _${i18n.get('/image', locale)}_`,
            '',
            `- \`/found\` \`/owned\`: _${i18n.get('/found', locale)}_`,
            '',
            `- \`/likeslist\`: _${i18n.get('/likeslist', locale)}_`,
            '',
            `- \`/like\` \`/protect\` \`/wish\`: _${i18n.get(
              '/like',
              locale
            )}_`,
            `- \`/unlike\`: _${i18n.get('/unlike', locale)}_`,
            `- \`/likeall\`: _${i18n.get('/likeall', locale)}_`,
            `- \`/unlikeall\`: _${i18n.get('/unlikeall', locale)}_`,
            '',
            `- \`/buy normal\` \`/shop normal\`: _${i18n.get(
              '/buy normal',
              locale
            )}_`,
            `- \`/buy guaranteed\` \`/shop guaranteed\`: _${i18n.get(
              '/buy guaranteed',
              locale
            )}_`,
            '',
            `- \`/automerge min\`: _${i18n.get('/automerge min', locale)}_`,
            `- \`/automerge max\`: _${i18n.get('/automerge max', locale)}_`,
            '',
            `- \`/installed packs\`: _${i18n.get('/installed packs', locale)}_`,
            '',
            `- \`/logs\` \`/history\`: _${i18n.get('/logs', locale)}_`,
          ].join('\n')
        )
    ),
    new discord.Message().addEmbed(
      new discord.Embed()
        .setAuthor({
          name: i18n.get('help-page9-title', locale),
        })
        .setDescription(
          [
            i18n.get('admin-permission', locale),
            `- \`/server options\`: _${i18n.get('/server options', locale)}_`,
            '',
            `- \`/packs install\`: _${i18n.get('/packs install', locale)}_`,
            `- \`/packs uninstall\`: _${i18n.get('/packs uninstall', locale)}_`,
            '',
            i18n.get('make-your-own-pack', locale),
          ].join('\n')
        )
    ),
  ];

  return discord.Message.page({
    message: pages[index],
    total: pages.length,
    type: 'help',
    next: index + 1 < pages.length,
    index,
    locale,
  });
}

const help = {
  pages,
};

export default help;
