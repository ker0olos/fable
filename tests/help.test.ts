import { describe, it, expect, beforeEach, vi } from 'vitest';

import help from '~/src/help.ts';

describe('/help', () => {
  beforeEach(() => {
    vi.restoreAllMocks();
  });

  it('navigation', () => {
    const message = help.pages({ userId: 'user_id', index: 0 });

    expect(message.json().data.components[0].components[0]).toEqual({
      custom_id: 'help==7=prev',
      label: 'Prev',
      style: 2,
      type: 2,
    });

    expect(message.json().data.components[0].components[1]).toEqual({
      custom_id: '_',
      disabled: true,
      label: '1/8',
      style: 2,
      type: 2,
    });

    expect(message.json().data.components[0].components[2]).toEqual({
      custom_id: 'help==1=next',
      label: 'Next',
      style: 2,
      type: 2,
    });
  });

  it('pages', () => {
    for (let i = 0; i < 8; i++) {
      const message = help.pages({ userId: 'user_id', index: i });

      expect(message.json()).toMatchSnapshot();
    }
  });
});
