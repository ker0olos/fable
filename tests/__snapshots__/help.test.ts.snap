export const snapshot = {};

snapshot[`/help 1`] = `
{
  "type": 4,
  "data": {
    "embeds": [
      {
        "type": "rich",
        "author": {
          "name": "1."
        },
        "title": "\`/gacha\`",
        "description": "Before anything: You will need characters.\\n\\n__You get a free pull every 30 minutes (maxed at 5 pulls)__. Use \`/now\` to check how many pulls you have and how much time is left until your next free pull.\\n\\n__Characters are exclusive__, once a player on this server finds a character, no one else can find that character in gacha again.\\n\\n__Characters vary in ranks from 1<:smolstar:1107503653956374638>to 5<:smolstar:1107503653956374638>__ A character rating is based on their popularity (a character rating cannot be raised).\\n\\n> *\`/q\` allows you to do gacha pulls with no animations*\\n​",
        "footer": {
          "text": "aliases: /w, /q"
        }
      }
    ],
    "attachments": [],
    "components": [
      {
        "type": 1,
        "components": [
          {
            "type": 2,
            "custom_id": "help==8=prev",
            "label": "Prev",
            "style": 2
          },
          {
            "type": 2,
            "custom_id": "_",
            "label": "1/9",
            "disabled": true,
            "style": 2
          },
          {
            "type": 2,
            "custom_id": "help==1=next",
            "label": "Next",
            "style": 2
          },
          {
            "type": 2,
            "custom_id": "gacha=user_id",
            "label": "/gacha",
            "style": 2
          },
          {
            "type": 2,
            "custom_id": "now=user_id",
            "label": "/now",
            "style": 2
          }
        ]
      },
      {
        "type": 1,
        "components": [
          {
            "type": 2,
            "label": "GitHub",
            "url": "https://github.com/ker0olos/fable",
            "style": 5
          },
          {
            "type": 2,
            "label": "Discord Support Server",
            "url": "https://discord.gg/H69RVBxeYY",
            "style": 5
          },
          {
            "type": 2,
            "label": "Donate",
            "url": "https://www.buymeacoffee.com/ker0olos",
            "style": 5
          }
        ]
      }
    ]
  }
}
`;

snapshot[`/help 2`] = `
{
  "type": 4,
  "data": {
    "embeds": [
      {
        "type": "rich",
        "author": {
          "name": "2."
        },
        "title": "\`/merge\`",
        "description": "Sacrifice characters to pull a new character with a higher rating.\\n\\n__Your party members and likes will never be sacrificed.__ To protect characters from being merged, add them to your likeslist with \`/like\` or \`/likeall\`\\n\\n__It takes 5 characters with a lower rating__ to pull a new character\\n\\n__Fable will always auto-merge your lowest-rated characters first.__\\n\\nExample: \`/merge target: 3\` will sacrifice __25 of your 1<:smolstar:1107503653956374638>characters__\\nor __5 of your 2<:smolstar:1107503653956374638>characters.__\\n\\n> *Sacrificed characters will return to being available in \`/gacha\`*\\n​",
        "footer": {
          "text": "aliases: /synthesis"
        }
      }
    ],
    "attachments": [],
    "components": [
      {
        "type": 1,
        "components": [
          {
            "type": 2,
            "custom_id": "help==0=prev",
            "label": "Prev",
            "style": 2
          },
          {
            "type": 2,
            "custom_id": "_",
            "label": "2/9",
            "disabled": true,
            "style": 2
          },
          {
            "type": 2,
            "custom_id": "help==2=next",
            "label": "Next",
            "style": 2
          }
        ]
      }
    ]
  }
}
`;

snapshot[`/help 3`] = `
{
  "type": 4,
  "data": {
    "embeds": [
      {
        "type": "rich",
        "author": {
          "name": "3."
        },
        "title": "\`/party\`",
        "description": "You will need a party to use combat features\\n\\n__You can't have more than 1 party__ or save your current party to a preset and switch between presets.\\n\\n\`/party view\` to see your current party.\\n\\n\`/party assign name: spot:\` to assign a character you have to your party,\\nleaving the \`spot:\` empty will assign the character to the first empty spot or override the last spot.\\n\\n> *\`/collection\` can be used to browse your characters and select your potential party members*\\n​",
        "footer": {
          "text": "aliases: /team, /p"
        }
      }
    ],
    "attachments": [],
    "components": [
      {
        "type": 1,
        "components": [
          {
            "type": 2,
            "custom_id": "help==1=prev",
            "label": "Prev",
            "style": 2
          },
          {
            "type": 2,
            "custom_id": "_",
            "label": "3/9",
            "disabled": true,
            "style": 2
          },
          {
            "type": 2,
            "custom_id": "help==3=next",
            "label": "Next",
            "style": 2
          },
          {
            "type": 2,
            "custom_id": "cstars=5=user_id",
            "label": "/collection stars 5",
            "style": 2
          },
          {
            "type": 2,
            "custom_id": "cstars=4=user_id",
            "label": "/collection stars 4",
            "style": 2
          }
        ]
      }
    ]
  }
}
`;

snapshot[`/help 4`] = `
{
  "type": 4,
  "data": {
    "embeds": [
      {
        "type": "rich",
        "author": {
          "name": "Stealing"
        },
        "description": "When all negotiations fails, it's fine to take what you want by force, right?\\n\\n__You can't steal characters assigned as party members__, meaning you can protect your most precious characters by adding them to your party.\\n\\n__Inactive users are easier to steal from.__ Stay active by using \`/gacha\` or \`/q\` once a day.\\n\\n__Inactive users lose their party protection__ after **4** days.\\n\\n__The higher the character's rating the harder it is to steal.__ 5<:smolstar:1107503653956374638>characters have a base success rate of 1%.\\n\\n__Steal has a cooldown of **3** days__ regardless of the outcome of the attempt.\\n\\n> Staying active is the best defense against stealing.\\n​"
      }
    ],
    "attachments": [],
    "components": [
      {
        "type": 1,
        "components": [
          {
            "type": 2,
            "custom_id": "help==2=prev",
            "label": "Prev",
            "style": 2
          },
          {
            "type": 2,
            "custom_id": "_",
            "label": "4/9",
            "disabled": true,
            "style": 2
          },
          {
            "type": 2,
            "custom_id": "help==4=next",
            "label": "Next",
            "style": 2
          }
        ]
      }
    ]
  }
}
`;

snapshot[`/help 5`] = `
{
  "type": 4,
  "data": {
    "embeds": [
      {
        "type": "rich",
        "author": {
          "name": "Combat"
        },
        "description": "__Every character gets a starting amount of stats and skills__ based on their star rating.\\n\\n__Stats and skills are randomly generated for every new character you find__, use \`/stats\` to check.\\n\\n__Attack is the damage__, that you can potentially do to your enemy.\\n\\n__Defense is your damage reduction__, minimizing the damage a enemy can do to you.\\n\\n__Speed is your turn order__, the faster between you and your enemy gets the first move, an overwhelming speed difference will allow you to attack move than once in the same turn.\\n\\nUse \`/battle tower\` to climb the tower, every server has a unique tower and unique enemies.\\n\\nLevel up your characters and gain extra stat points by \`/reclear\`ing the same floor a few times.\\n\\n> Stat points gained from leveling up is random distributed based on the ratio of your starting stats, Skill points can be used to acquire or upgrade \`/skills\`\\n\\n> Challenging and reclearing floors require keys, you get +1 key every 10 minutes, you can also buy more keys from the \`/shop\`\\n​"
      }
    ],
    "attachments": [],
    "components": [
      {
        "type": 1,
        "components": [
          {
            "type": 2,
            "custom_id": "help==3=prev",
            "label": "Prev",
            "style": 2
          },
          {
            "type": 2,
            "custom_id": "_",
            "label": "5/9",
            "disabled": true,
            "style": 2
          },
          {
            "type": 2,
            "custom_id": "help==5=next",
            "label": "Next",
            "style": 2
          }
        ]
      }
    ]
  }
}
`;

snapshot[`/help 6`] = `
{
  "type": 4,
  "data": {
    "embeds": [
      {
        "type": "rich",
        "author": {
          "name": "Shop"
        },
        "description": "The shop is a way to purchase extra pulls and guaranteed pulls.\\n\\n__You get 1 daily token every 12 hours__, tokens are added automatically. \\n\\nYou get 2 daily tokens on weekends.\\n\\n\`/buy normal\` gets you extra pulls, they are the exact same as the ones you get every 30 minutes.\\n\\n\`/buy guaranteed\` guarantees you get a character from one specific star rating.\\n\\nExample: \`/buy guaranteed 5\` will guarantee you get one random 5<:smolstar:1107503653956374638>character.\\n\\n3<:smolstar:1107503653956374638>cost **4** tokens\\n4<:smolstar:1107503653956374638>cost **12** tokens\\n5<:smolstar:1107503653956374638>cost **28** tokens\\n\\n> Use the guaranteed pulls you bought by calling \`/pull stars:\`\\n​",
        "footer": {
          "text": "aliases: /shop"
        }
      }
    ],
    "attachments": [],
    "components": [
      {
        "type": 1,
        "components": [
          {
            "type": 2,
            "custom_id": "help==4=prev",
            "label": "Prev",
            "style": 2
          },
          {
            "type": 2,
            "custom_id": "_",
            "label": "6/9",
            "disabled": true,
            "style": 2
          },
          {
            "type": 2,
            "custom_id": "help==6=next",
            "label": "Next",
            "style": 2
          },
          {
            "type": 2,
            "custom_id": "buy=bguaranteed=user_id=5",
            "label": "/buy guaranteed 5",
            "style": 2
          }
        ]
      }
    ]
  }
}
`;

snapshot[`/help 7`] = `
{
  "type": 4,
  "data": {
    "embeds": [
      {
        "type": "rich",
        "author": {
          "name": "Essential Commands"
        },
        "description": "- \`/gacha\` \`/w\`: _start a new gacha pull_\\n- \`/q\`: _start a gacha pull but with no animations_\\n- \`/pull\` \`/guaranteed\`: _use a guaranteed pull you have_\\n\\n- \`/now\` \`/tu\`: _check what you can do right now_\\n- \`/search\` \`/anime\` \`/manga\` \`/series\`: _search for a specific series_\\n- \`/character\` \`/char\`: _search for a specific character_\\n\\n- \`/merge\` \`/synthesize\`: _sacrifice characters to pull a new more shiny character_\\n\\n- \`/party view\` \`/team view\` \`/p view\`: _view your current party_\\n- \`/party assign\` \`/team assign\` \`/p assign\`: _assign a character to your party_\\n- \`/party swap\` \`/team swap\` \`/p swap\`: _swap characters spots with each others_\\n- \`/party remove\` \`/team remove\` \`/p remove\`: _remove a character from your party_\\n- \`/party clear\` \`/team clear\` \`/p clear\`: _remove all characters from your party_\\n\\n- \`/collection show\` \`/coll show\` \`/mm show\`: _check and view how complete are a user collections_\\n- \`/collection stars\` \`/coll stars\` \`/mm stars\`: _view user stars_\\n- \`/collection media\` \`/coll media\` \`/mm media\`: _view user characters in a specific series_\\n- \`/collection sum\` \`/coll sum\` \`/mm sum\`: _view the sum numbers of a user collection_\\n\\n- \`/steal\`: _steal a character from another user_\\n- \`/trade\` \`/offer\`: _trade characters with another user_\\n- \`/give\` \`/gift\`: _give characters to another user_\\n\\n- \`/battle tower\`: _Climb the tower to glory and become a fable_\\n- \`/battle challenge\`: _Challenge the tower's next floor_\\n- \`/reclear\`: _Reclear your current tower floor to gain exp_"
      }
    ],
    "attachments": [],
    "components": [
      {
        "type": 1,
        "components": [
          {
            "type": 2,
            "custom_id": "help==5=prev",
            "label": "Prev",
            "style": 2
          },
          {
            "type": 2,
            "custom_id": "_",
            "label": "7/9",
            "disabled": true,
            "style": 2
          },
          {
            "type": 2,
            "custom_id": "help==7=next",
            "label": "Next",
            "style": 2
          }
        ]
      }
    ]
  }
}
`;

snapshot[`/help 8`] = `
{
  "type": 4,
  "data": {
    "embeds": [
      {
        "type": "rich",
        "author": {
          "name": "Other Commands"
        },
        "description": "- \`/nick\`: _change the nickname of a character_\\n- \`/image\` \`/custom\`: _change the image of a character_\\n\\n- \`/found\` \`/owned\`: _list all characters found from a specific series_\\n\\n- \`/likeslist\`: _list all user likes_\\n\\n- \`/like\` \`/protect\` \`/wish\`: _like a character to be notified if someone finds them_\\n- \`/unlike\`: _remove character from your likes_\\n- \`/likeall\`: _like a series to be notified if someone finds any character from it_\\n- \`/unlikeall\`: _remove a series from your likes_\\n\\n- \`/buy normal\` \`/shop normal\`: _use tokens to buy normal pulls_\\n- \`/buy guaranteed\` \`/shop guaranteed\`: _use tokens to buy pulls with a specific guaranteed rating_\\n\\n- \`/automerge min\`: _merge your characters together to pull the lowest rating possible_\\n- \`/automerge max\`: _merge your characters together to pull the highest rating possible_\\n\\n- \`/installed packs\`: _list all the installed packs on a server_\\n\\n- \`/logs\` \`/history\`: _list user last few found characters_"
      }
    ],
    "attachments": [],
    "components": [
      {
        "type": 1,
        "components": [
          {
            "type": 2,
            "custom_id": "help==6=prev",
            "label": "Prev",
            "style": 2
          },
          {
            "type": 2,
            "custom_id": "_",
            "label": "8/9",
            "disabled": true,
            "style": 2
          },
          {
            "type": 2,
            "custom_id": "help==8=next",
            "label": "Next",
            "style": 2
          }
        ]
      }
    ]
  }
}
`;

snapshot[`/help 9`] = `
{
  "type": 4,
  "data": {
    "embeds": [
      {
        "type": "rich",
        "author": {
          "name": "Admin Commands"
        },
        "description": "_> Those commands require the \`Manage Server\` permission by default, but that can be changed in the \`Integrations\` section of your server settings, although we never recommend that._\\n- \`/server options\`: _manage your server_\\n\\n- \`/packs install\`: _install a community pack_\\n- \`/packs uninstall\`: _uninstall a community pack_\\n\\n> To create your own community pack visit <https://packs.fable.ing>"
      }
    ],
    "attachments": [],
    "components": [
      {
        "type": 1,
        "components": [
          {
            "type": 2,
            "custom_id": "help==7=prev",
            "label": "Prev",
            "style": 2
          },
          {
            "type": 2,
            "custom_id": "_",
            "label": "9/9",
            "disabled": true,
            "style": 2
          },
          {
            "type": 2,
            "custom_id": "help==0=next",
            "label": "Next",
            "style": 2
          }
        ]
      }
    ]
  }
}
`;
